components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/AMDefaultKeyManagerImpl.java

```diff
@@ -756,15 +748,28 @@ public void loadConfiguration(KeyManagerConfiguration configuration) throws APIM
                 .requestInterceptor(new TenantHeaderInterceptor(tenantDomain))
                 .errorDecoder(new KMClientErrorDecoder())
                 .target(UserClient.class, userInfoEndpoint);
-        revokeClient = Feign.builder()
-                .client(new ApacheFeignHttpClient(APIUtil.getHttpClient(revokeOneTimeTokenEndpoint)))
-                .encoder(new GsonEncoder())
-                .decoder(new GsonDecoder())
-                .logger(new Slf4jLogger())
-                .requestInterceptor(new BasicAuthRequestInterceptor(username, password))
-                .requestInterceptor(new TenantHeaderInterceptor(tenantDomain))
-                .errorDecoder(new KMClientErrorDecoder())
-                .target(RevokeClient.class, revokeOneTimeTokenEndpoint);
+
+        if (!configuration.getType().equals(APIConstants.KeyManager.WSO2_IS_KEY_MANAGER_TYPE)) {
```

Comment:
Shall we interchange the sides of the comparing parameters in equals? This would prevent a possible null pointer exception.